src/redux/features/votes/votesSlice.ts

```diff
@@ -0,0 +1,51 @@
+import { ActionReducerMapBuilder, PayloadAction, createSlice } from '@reduxjs/toolkit';
+
+import { StatusT, buildReducers } from 'src/redux/features/asyncUtils';
+import {
+  requestProjectVotesDelete,
+  requestProjectVotesGet,
+  requestProjectVotesUpdate,
+} from 'src/redux/features/votes/votesAsyncThunks';
+import { GetProjectVotesResponse, GetProjectVotesResponsePayload } from 'src/types/Votes';
+
+// Define a type for the slice state
+type Data = {
+  error?: string;
+  votes?: GetProjectVotesResponsePayload;
+};
+
+// Define the initial state
+const initialState: Data = {};
+
+export const votesSlice = createSlice({
+  name: 'votesSlice',
+  initialState,
+  reducers: {
+    setProjectVotesAction: (state, action: PayloadAction<Data>) => {
+      const data: Data = action.payload;
+      state.error = data.error;
+      state.votes = data.votes;
+    },
+  },
+});
+
+export const { setProjectVotesAction } = votesSlice.actions;
+export const votesReducer = votesSlice.reducer;
+
+type ProjectVotesResponsesUnion = GetProjectVotesResponse;
+type VotingRequestsState = Record<string, StatusT<ProjectVotesResponsesUnion>>;
+
+const initialVotingRequestsState: VotingRequestsState = {};
+
+export const votesRequestsSlice = createSlice({
+  name: 'votesRequestsSlice',
+  initialState: initialVotingRequestsState,
+  reducers: {},
+  extraReducers: (builder: ActionReducerMapBuilder<VotingRequestsState>) => {
+    buildReducers(requestProjectVotesGet)(builder);
```

Comment:
I'm assuming when we edit or delete votes, we only want to know if it succeeded (boolean true/false).
That is the pattern so far, the client refreshes the view after that.
Votes Update/Delete should have a result store of requestId to boolean status, ideally.
The get project votes should be a separate slice on it's own, keyed off the projectId.

Reply:
Ok that makes sense that we'd want to store data differently for those two types of requests, thank you for explaining. Will update.

Reply:
@karthikbtf so actually I just realized the delete response is boolean only but the upsert response returns some data. In that case how should we handle? Split all three? 🤔

Reply:
My thoughts but will leave it to you. For now we can keep the delete and update as booleans, the client will just reload all votes. It's a pattern we use everywhere. At some point if we need to get more efficient, we can split them down the road.